garbage_collector.rst

```diff
@@ -0,0 +1,460 @@
+generations are. These thresholds can be examined using the  ``gc.get_threshold``
+function:
+
+.. code-block:: python
+
+    >>> import gc
+    >>> gc.get_threshold()
+    (700, 10, 10)
+
+
+The content of these generations can be examined using the
+``gc.get_objects(generation=NUM)`` function and collections can be triggered
+specifically in a generation by calling ``gc.collect(generation=NUM)``.
+
+.. code-block:: python
+
+    >>> import gc
+    >>> class MyObj:
+    ...     pass
+    ... 
+
+    # Move everything to the last generation so its easier to inspect
+    # the younger generations.
+
+    >>> gc.collect()
+    0
+
+    # Create a reference cycle
+
+    >>> x = MyObj()
+    >>> x.self = x
+
+    # Initially the object is in the younguest generation.
+
+    >>> gc.get_objects(generation=0)
+    [..., <__main__.MyObj object at 0x7fbcc12a3400>, ...]
+
+    # After a collection of the younguest generation the object
+    # moves to the next generation.
+
+    >>> gc.collect(generation=0)
+    0
+    >>> gc.get_objects(generation=0)
+    []
+    >>> gc.get_objects(generation=1)
+    [..., <__main__.MyObj object at 0x7fbcc12a3400>, ...]
+
+
+
+Collecting the oldest generation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to the various configurable thresholds, the GC only triggers a full
+collection of the oldest generation if the ratio ``long_lived_pending / long_lived_total``
+is above a given value (hardwired to 25%). The reason is that, while "non-full"
+collections (i.e., collections of the young and middle generations) will always
+examine roughly the same number of objects (determined by the aforementioned
+thresholds) the cost of a full collection is proportional to the total
+number of long-lived objects, which is virtually unbounded.  Indeed, it has
+been remarked that doing a full collection every <constant number> of object
+creations entails a dramatic performance degradation in workloads which consist
+of creating and storing lots of long-lived objects (e.g. building a large list
+of GC-tracked objects would show quadratic performance, instead of linear as
+expected). Using the above ratio, instead, yields amortized linear performance
+in the total number of objects (the effect of which can be summarized thusly:
+"each full garbage collection is more and more costly as the number of objects
+grows, but we do fewer and fewer of them").
+
+Optimization: reusing fields to save memory
+-------------------------------------------
+
+In order to save memory, the two linked list pointers in every object with GC
+support are reused for several purposes. This is a common optimization known
+as "fat pointers" or "tagged pointers": pointers that carry additional data,
+"folded" into the pointer, meaning stored inline in the data representing the
+address, taking advantage of certain properties of memory addressing. This is
+possible as most architectures are certain types of data will often be aligned
```

Comment:
Something is missing or extraneous here, or the order is wrong.  Is the following what you meant?
```suggestion
possible as most architectures align certain types of data
```